packages/components/src/DataList/context/DataListSortContext/DataListSortContext.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { DataListSortContextProps, Sorting } from "../../DataList.types";
+import { useDataListContext } from "../DataListContext";
+
+const DataListSortContext = createContext<DataListSortContextProps>({
+  sortingState: {
+    direction: "none",
+    key: "",
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  toggleSorting: (_: string) => {
+    return;
+  },
+});
+
+export function useDataListSortContext() {
+  return useContext(DataListSortContext);
+}
+
+export function DataListSortProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const { sorting: sortingProp } = useDataListContext();
+
+  const [sortingState, setSortingState] = useState<Sorting>(
+    sortingProp?.initialState || {
+      direction: "none",
+      key: "",
+    },
+  );
```

Comment:
best practice on this is to never track the state of a component that could be updated outside of it. You risk making the component does something different than what is in the main state that the consumer uses.

Also easy to spot it of it's broken. If you click it and it does nothing, you, the consumer hasn't implemented it properly. Compared to it updating the state internally but actually not the main state outside of it. 

In short, keep it to be one source of truth instead of 2